File: solutions/Pairwise-Linked-List-Swap.java
import java.util.*;

/**
 * class LLNode {
 *   int val;
 *   LLNode next;
 * }
 */
class Solution {
    public LLNode solve(LLNode node) {
        for (LLNode curr = node; curr != null && curr.next != null; curr = curr.next.next) {
            int tmp = curr.val;
            curr.val = curr.next.val;
            curr.next.val = tmp;
        }
        return node;
    }
}

File: solutions/Pairwise-Linked-List-Swap.cpp
/**
 * class LLNode {
 *     public:
 *         int val;
 *         LLNode *next;
 * };
 */
LLNode *solve(LLNode *node)
{
    if (!node || !(node->next))
        return node;

    // Let a -> b -> (c1 -> c2 -> ...)
    // be  a -> b ->  c
    LLNode *a = node;
    LLNode *b = node->next;
    LLNode *c = solve(b->next);

    // we want b -> a -> c
    b->next = a;
    a->next = c;

    return b;
}
